src/ol_concourse/lib/resources.py

```diff
@@ -257,3 +257,19 @@ def slack_notification(name: Identifier, url: str) -> Resource:
     return Resource(
         name=name, type="slack-notification", source={"url": url, "disabled": False}
     )
+
+
+def npm_package(package: Identifier, scope: Identifier) -> Resource:
+    return Resource(
+        name=package,
+        type="npm-package",
+        icon="language-javascript",
+        source={
+            "package": package,
+            "scope": scope,
+            "registry": {
+                "uri": "https://registry.npmjs.org",
+                "token": "{{npmjs.concourse_token}}",
```

Comment:
I don't think this is right. Refer to lines 214 + 215 above for how it is done with pypi. https://github.com/mitodl/ol-infrastructure/pull/2303/files#diff-ab9c861e659afe118ba16db129458952fb2ddbf8f5c3104c58864b89b1733c1cR215-R216. 

I also don't see the secret(s) in vault production?  You can add it to vault automatically by putting it in the sops file and then this block in the concourse pulumi stack will take care of it: https://github.com/mitodl/ol-infrastructure/pull/2303/files#diff-ab9c861e659afe118ba16db129458952fb2ddbf8f5c3104c58864b89b1733c1cR215-R216

It would be something like:

```yaml
pipelines:
    <team name that needs the token>/npmjs:
        token: <your super secret token> 
```

And then this line would be:

```
"token": "((npmjs.token))",
```

It will make more sense once you decrypt that file. You will need to do an entry for each team that needs it. _Most_ of our stuff is in the `infrastructure` team.